core/src/main/java/org/parchmentmc/feather/util/CollectorUtils.java

```diff
@@ -0,0 +1,168 @@
+     * order, using {@link Collectors#toConcurrentMap(Function, Function, BinaryOperator)}
+     * may offer better parallel performance.
+     * <p>
+     * The returned {@code Map} is linked and will retain its insertion order.
+     *
+     * @param <T> the type of the input elements
+     * @param <K> the output type of the key mapping function
+     * @param <U> the output type of the value mapping function
+     * @param keyMapper a mapping function to produce keys
+     * @param valueMapper a mapping function to produce values
+     * @param mergeFunction a merge function, used to resolve collisions between
+     *                      values associated with the same key, as supplied
+     *                      to {@link Map#merge(Object, Object, BiFunction)}
+     * @return a {@code Collector} which collects elements into a {@code Map}
+     * whose keys are the result of applying a key mapping function to the input
+     * elements, and whose values are the result of applying a value mapping
+     * function to all input elements equal to the key and combining them
+     * using the merge function
+     *
+     * @see Collectors#toMap(Function, Function)
+     * @see Collectors#toMap(Function, Function, BinaryOperator, Supplier)
+     * @see Collectors#toConcurrentMap(Function, Function, BinaryOperator)
+     */
+    public static <T, K, U>
+    Collector<T, ?, Map<K,U>> toLinkedMap(Function<? super T, ? extends K> keyMapper,
+                                    Function<? super T, ? extends U> valueMapper,
+                                    BinaryOperator<U> mergeFunction) {
+        return Collectors.toMap(keyMapper, valueMapper, mergeFunction, LinkedHashMap::new);
+    }
+
+    /**
+     * Returns a merge function, suitable for use in
+     * {@link Map#merge(Object, Object, BiFunction) Map.merge()} or
+     * {@link Collectors#toMap(Function, Function, BinaryOperator) toMap()}, which always
+     * throws {@code IllegalStateException}.  This can be used to enforce the
+     * assumption that the elements being collected are distinct.
+     *
+     * @param <T> the type of input arguments to the merge function
+     * @return a merge function which always throw {@code IllegalStateException}
+     */
+    private static <T> BinaryOperator<T> throwingMerger() {
+        return (u,v) -> { throw new IllegalStateException(String.format("Duplicate key %s", u)); };
+    }
+
+    /**
+     * Returns a {@code Collector} that accumulates the input elements into a
+     * new {@code Set}. There are no guarantees on the type, mutability,
+     * serializability, or thread-safety of the {@code Set} returned, however insertion
+     * order is maintained.
+     *
+     * @param <T> the type of the input elements
+     * @return a {@code Collector} which collects all the input elements into a
+     * {@code Set} maintaining insertion order
+     */
+    public static <T> Collector<T, ?, Set<T>> toLinkedSet() {
```

Comment:
A quick search for uses of the JDK `Collectors` class show that there are still calls to `Collectors.toCollection(LinkedHashSet::new)`. Those should be all replaced with calls to this method.